src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,413 @@
+        printWelcome();
+        runMainMenu();
+    }
+
+    /**
+     * Prints the welcome message which includes logo.
+     */
+    private static void printWelcome() {
+        System.out.println(CAT_LOGO);
+
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println(" Hello! I'm Neko-bot *meow*");
+        System.out.println(" What can I do for you?");
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    /**
+     * Runs main menu for the program. Does a while loop until user inputs bye.
+     */
+    private static void runMainMenu() {
+        boolean isProcessingCommand = true;
+        ArrayList<Task> taskList = new ArrayList<>();
+
+        Scanner userInput = new Scanner(System.in);
+
+        try {
+            taskList = readLocalList();
+            if (taskList.size() > 0) {
+                System.out.println("Successfully loaded " + taskList.size() + " tasks from previous session.");
+            } else {
+                System.out.println("Looks like you're new here, starting with a fresh task list!");
+            }
+            System.out.println(HORIZONTAL_LINE);
+        } catch (IOException e) {
+            e.printStackTrace();
+            System.out.println("File IO exception, we had difficulty managing your local task file.");
+        }
+
+
+        while (isProcessingCommand) {
+            try {
+                isProcessingCommand = processCommand(taskList, userInput);
+            } catch (DukeException exceptionMessage) {
+                exceptionMessage.printStackTrace();
+            }
+
+        }
+    }
+
+
+    /**
+     * Processes user input from main menu.
+     *
+     * @param taskList  ArrayList of all the tasks.
+     * @param userInput User input string.
+     * @return Returns true unless user enters bye, which would terminate main menu.
+     */
+    private static boolean processCommand(ArrayList<Task> taskList, Scanner userInput) throws DukeException {
+        String input = userInput.nextLine();
+
+        // Split string into 2's using space
+        String[] splitInput = input.split(REGEX_SINGLE_SPACE, 2);
+        String taskDescription = processSplitString(splitInput);
+
+        // Main menu navigation
+        switch (splitInput[0]) {
+        case COMMAND_BYE:
+            printGoodbye();
+            return false;
+        case COMMAND_LIST:
+            processListCommand(taskList);
+            break;
+        case COMMAND_DONE:
+            processDoneCommand(taskList, taskDescription);
+            break;
+        case COMMAND_DELETE:
+            processDeleteCommand(taskList, taskDescription);
+            break;
+        case COMMAND_TODO:
+            processTodoCommand(taskList, taskDescription);
+            break;
+        case COMMAND_DEADLINE:
+            processDeadlineCommand(taskList, taskDescription);
+            break;
+        case COMMAND_EVENT:
+            processEventCommand(taskList, taskDescription);
```

Comment:
You may want to consider extracting these commands out as classes instead of methods to adopt an OOP design.